lib/elixir/lib/module/parallel_checker.ex

```diff
@@ -309,24 +309,51 @@ defmodule Module.ParallelChecker do
     Enum.flat_map(warnings, fn {module, warning, locations} ->
       message = module.format_warning(warning)
       diagnostics = Enum.map(locations, &to_diagnostic(message, &1))
-      log? and :elixir_errors.print_warning([message, ?\n, format_stacktraces(diagnostics)])
+      log? and print_warning(message, diagnostics)
       diagnostics
     end)
   end
 
-  defp format_stacktraces([diagnostic]) do
-    format_diagnostic_stacktrace(diagnostic)
+  defp print_warning(message, [diagnostic]) do
+    :elixir_errors.print_warning(message, diagnostic)
   end
 
-  defp format_stacktraces(diagnostics) do
-    [
-      "Invalid call found at #{length(diagnostics)} locations:\n",
-      Enum.map(diagnostics, &format_diagnostic_stacktrace/1)
-    ]
-  end
+  defp print_warning(message, [first_diagnostic | rest]) do
+    :elixir_errors.print_warning(message, first_diagnostic)
+
+    %{position: position, file: file} = first_diagnostic
+
+    line =
+      case position do
+        {line, _} -> line
+        line -> line
+      end
+
+    line_digits = :elixir_errors.get_line_number_digits(line, 1)
+
+    # Match indentation with the formatted warning
+    # When we have the line available, we add 4 spaces, else 2
+    stacktrace_padding =
+      if File.regular?(file) do
+        line_digits + 4
+      else
+        line_digits + 2
+      end
+
+    warn_locations =
+      for diagnostic <- rest do
+        %{stacktrace: [s]} = diagnostic
+        String.duplicate(" ", stacktrace_padding) <> Exception.format_stacktrace_entry(s) <> "\n"
+      end
+
+    locations_plural = (length(warn_locations) == 1 && "location") || "locations"
+
+    rest =
+      String.duplicate(" ", stacktrace_padding - 2) <>
+        "Invalid call also found at #{length(warn_locations)} other #{locations_plural}:\n" <>
+        :erlang.list_to_binary(warn_locations)
```

Comment:
We want to make all of this part of the message, instead of printing separately, because if it is in the message, then `mix compile --all-warnings` (which is the default) will print it again with all locations.

Reply:
If we make it part of the message, we don't need to call `:elixir_errors.get_line_number_digits(line, 1)` and friends either.

Reply:
> If we make it part of the message, we don't need to call :elixir_errors.get_line_number_digits(line, 1) and friends either.

Sorry, just saw this one now.

I made it part of the message in 32c44eddf139ece83dbe5791d7f998660475ced3 but I don't think we can avoid calling those functions from `:elixir_errors` if doing it on the elixir since, since their output is used to build the grouped warnings message

Reply:
Sorry, to be clear, I think this code would be fine:


```
  defp print_warning(message, [diagnostic]) do
    :elixir_errors.print_warning(message, diagnostic)
  end

  defp print_warning(message, [diagnostic | rest]) do
    locations =
      Enum.map(rest, fn %{stacktrace: [stacktrace]} ->
        ["  ", Exception.format_stacktrace_entry(stacktrace), ?\n]
      end)

    [
      message,
      "\n\nSimilar warning found at #{length(rest)} other locations:\n",
      locations
    ]
    |> IO.iodata_to_binary()
    |> :elixir_errors.print_warning(diagnostic)
  end
```

Reply:
That's very nice! I thought about the formatted snippet rather than the message parameter.

I've updated the code to use this instead!

I see you changed the warning message here to `"Similar warning found at..."` rather than `"invalid call also found at..."`. 
I didn't update it here but I think it makes sense to use the warning term instead of invalid call, I can open another PR to change it if you want

Reply:
Yes, let's update it in another PR. :)